src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/YurinekoParser.kt

```diff
@@ -0,0 +1,151 @@
+package org.koitharu.kotatsu.parsers.site.vi
+
+import org.json.JSONObject
+import org.koitharu.kotatsu.parsers.MangaLoaderContext
+import org.koitharu.kotatsu.parsers.MangaSourceParser
+import org.koitharu.kotatsu.parsers.PagedMangaParser
+import org.koitharu.kotatsu.parsers.config.ConfigKey
+import org.koitharu.kotatsu.parsers.model.ContentType
+import org.koitharu.kotatsu.parsers.model.Manga
+import org.koitharu.kotatsu.parsers.model.MangaChapter
+import org.koitharu.kotatsu.parsers.model.MangaPage
+import org.koitharu.kotatsu.parsers.model.MangaSource
+import org.koitharu.kotatsu.parsers.model.MangaState
+import org.koitharu.kotatsu.parsers.model.MangaTag
+import org.koitharu.kotatsu.parsers.model.RATING_UNKNOWN
+import org.koitharu.kotatsu.parsers.model.SortOrder
+import org.koitharu.kotatsu.parsers.util.domain
+import org.koitharu.kotatsu.parsers.util.generateUid
+import org.koitharu.kotatsu.parsers.util.json.asIterable
+import org.koitharu.kotatsu.parsers.util.json.getStringOrNull
+import org.koitharu.kotatsu.parsers.util.json.mapJSON
+import org.koitharu.kotatsu.parsers.util.json.mapJSONToSet
+import org.koitharu.kotatsu.parsers.util.json.toJSONList
+import org.koitharu.kotatsu.parsers.util.mapChapters
+import org.koitharu.kotatsu.parsers.util.parseHtml
+import org.koitharu.kotatsu.parsers.util.parseJson
+import org.koitharu.kotatsu.parsers.util.parseJsonArray
+import org.koitharu.kotatsu.parsers.util.requireElementById
+import org.koitharu.kotatsu.parsers.util.toAbsoluteUrl
+import org.koitharu.kotatsu.parsers.util.tryParse
+import org.koitharu.kotatsu.parsers.util.urlEncoded
+import java.text.SimpleDateFormat
+import java.util.EnumSet
+import java.util.Locale
+
+@MangaSourceParser("YURINEKO", "Yurineko", "vi", ContentType.HENTAI)
+class YurinekoParser(context: MangaLoaderContext) : PagedMangaParser(context, MangaSource.YURINEKO, 20) {
+	override val configKeyDomain: ConfigKey.Domain
+		get() = ConfigKey.Domain("yurineko.net")
+
+	override val sortOrders: Set<SortOrder>
+		get() = EnumSet.of(SortOrder.UPDATED)
+
+	private val apiDomain = "api.$domain"
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val response = webClient.httpGet(manga.url.toAbsoluteUrl(apiDomain)).parseJson()
+		val df = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US)
+		return manga.copy(
+			chapters = response.getJSONArray("chapters")
+				.toJSONList()
+				.mapChapters(true) { i, jo ->
+					val mangaId = jo.getInt("mangaID")
+					val chapterId = jo.getInt("id")
+					MangaChapter(
+						id = generateUid(chapterId.toLong()),
+						name = jo.getString("name"),
+						number = i + 1,
+						scanlator = null,
+						url = "/read/$mangaId/$chapterId",
+						uploadDate = df.tryParse(jo.getString("date")),
+						branch = null,
+						source = source,
+					)
+				}.reversed(),
```

Comment:
You already passed `reverse=true` to the `mapChapters` function

Reply:
Thanks, I forgot to remove it